lib/param_parsing/versiontf.go

```diff
@@ -77,3 +78,9 @@ func isTerraformModule(params Params) bool {
 	module, err := tfconfig.LoadModule(params.ChDirPath)
 	return err == nil && len(module.RequiredCore) > 0
 }
+
+func cleanupVersionConstraints(constraint string) string {
+	regex := regexp.MustCompile(`(?P<Comparator>\D+)(?P<Version>(\d+\.\d+\.\d+)(-[a-zA-z]+\d*)?)$`)
```

Comment:
@MatrixCrawler First of all thanks for taking care of it — I indeed missed "improperly" formatted version constraints where constraint operator isn't separated from the version =(
My point is looking up `(\d+\.\d+\.\d+)` is incorrect since version can be specified as `major`, `major.minor`, and `major.minor.patch` (with an optional suffix that should fall under semver though). Hence we probably should better look up constraint operators instead: https://developer.hashicorp.com/terraform/language/expressions/version-constraints#version-constraint-syntax — `=`, `!=`, `>`, `>=`, `<`, `<=`, `~>`. And ideally follow what TF Config Inspect does: https://github.com/hashicorp/terraform-config-inspect/blob/master/tfconfig/load_hcl.go#L71-L78 (didn't go deeper though)

Reply:
I use the hashicorp version constraint checker now. no need for manually checking.

Reply:
Ah, just saw the ce0909ec0b8abb0d5478a944e1eabd1e23e7fcdc 👍🏻